Lib/test/test_imaplib.py

```diff
@@ -497,6 +497,56 @@ def test_with_statement_logout(self):
 
     # command tests
 
+    def test_idle_capability(self):
+        client, _ = self._setup(SimpleIMAPHandler)
+        with self.assertRaisesRegex(imaplib.IMAP4.error,
+                'does not support IDLE'):
+            with client.idle():
+                pass
+
+    class IdleCmdHandler(SimpleIMAPHandler):
+        capabilities = 'IDLE'
+        def cmd_IDLE(self, tag, args):
+            self._send_textline('+ idling')
+            self._send_line(b'* 2 EXISTS')
+            self._send_line(b'* 0 RECENT')
+            time.sleep(1)
+            self._send_line(b'* 1 RECENT')
+            r = yield
+            if r == b'DONE\r\n':
+                self._send_tagged(tag, 'OK', 'Idle completed')
+            else:
+                self._send_tagged(tag, 'BAD', 'Expected DONE')
+
+    def test_idle_iter(self):
+        client, _ = self._setup(self.IdleCmdHandler)
```

Comment:
I'm not a fan of using `_` here. Even if the developer knows what the value *should be*, that's exactly what tests are for -- making sure that value is correct.

Reply:
Oh! I didn't realize that. Nevermind then!